encord/storage.py

```diff
@@ -833,6 +833,70 @@ def add_private_data_to_folder_start(
         """
         return self._add_data_to_folder_start(integration_id, private_files, ignore_errors)
 
+    def sync_private_data_with_cloud_synced_folder_start(self) -> UUID:
+        sync_job_uuid = self._api_client.post(
+            path=f"storage/folders/{self.uuid}/cloud-synced-folder-syncs",
+            params=None,
+            payload=None,
+            result_type=UUID,
+            allow_retries=False,
+        )
+
+        logger.info(f"sync_private_data_with_cloud_synced_folder job started with {sync_job_uuid=}.")
+        logger.info("SDK process can be terminated, this will not affect successful job execution.")
+        logger.info("You can follow the progress in the web app via notifications.")
+
+        return sync_job_uuid
+
+    def sync_private_data_with_cloud_synced_folder_get_result(
+        self,
+        sync_job_uuid: UUID,
+        timeout_seconds: int = 7 * 24 * 60 * 60,  # 7 days
+    ) -> orm_storage.SyncPrivateDataWithCloudSyncedFolderGetResultResponse:
+        failed_requests_count = 0
+        polling_start_timestamp = time.perf_counter()
+
+        while True:
+            try:
+                res = self._api_client.get(
+                    f"storage/folders/{self.uuid}/cloud-linked-folder-syncs/{sync_job_uuid}/long-polling",
+                    params=orm_storage.SyncPrivateDataWithCloudSyncedFolderGetResultParams(
+                        timeout_seconds=timeout_seconds,
+                    ),
+                    result_type=orm_storage.SyncPrivateDataWithCloudSyncedFolderGetResultResponse,
+                )
+
+                if res.status == orm_storage.SyncPrivateDataWithCloudSyncedFolderStatus.DONE:
+                    logger.info(f"sync_private_data_with_cloud_synced_folder job completed with {sync_job_uuid=}.")
+
+                polling_elapsed_seconds = ceil(time.perf_counter() - polling_start_timestamp)
+                polling_available_seconds = max(0, timeout_seconds - polling_elapsed_seconds)
+
+                if (polling_available_seconds == 0) or (
```

Comment:
```suggestion
                if (polling_available_seconds <= 0) or (
```
would be a bit safer (I dunno, leap seconds are a thing...)